teamcity: failed test: TestRandomKeyAndTimestampExport

The following tests appear to have failed on master (testrace): TestRandomKeyAndTimestampExport/ts_(0-∞],_all,_nontimebound, TestRandomKeyAndTimestampExport/ts_(0-∞],_latest,_timebound, TestRandomKeyAndTimestampExport, TestRandomKeyAndTimestampExport/ts_(0-∞],_latest,_nontimebound

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestRandomKeyAndTimestampExport).

[#1514620](https://teamcity.cockroachdb.com/viewLog.html?buildId=1514620):


```
TestRandomKeyAndTimestampExport/ts_(0-∞],_latest,_timebound
...l.assertEqualKVs.func1(0xc004074600)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/export_test.go:328 +0x572 fp=0xc0036cbf70 sp=0xc0036cbac0 pc=0x46e0fa2
testing.tRunner(0xc004074600, 0xc0035301b0)
	/usr/local/go/src/testing/testing.go:865 +0x164 fp=0xc0036cbfd0 sp=0xc0036cbf70 pc=0xdde9b4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc0036cbfd8 sp=0xc0036cbfd0 pc=0xcdfa01
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x65b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0001b1600, 0x59ae402, 0x1f, 0x5acf4b0, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc0001b1600)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc0001b1600, 0xc000749d68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc0002ee4a0, 0x934e5a0, 0x11, 0x11, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc000630180, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestMain(0xc000630180)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/main_test.go:28 +0x15e
main.main()
	_testmain.go:82 +0x223

goroutine 6 [syscall]:
os/signal.signal_recv(0xcdfa01)
	/usr/local/go/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:29 +0x4f

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1222 +0x14f
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:625 +0x1e4

goroutine 8 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:632 +0x14b
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:626 +0x1fc

goroutine 13 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc00005f3c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:152 +0x1b3
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:29 +0x9a

goroutine 2091 [chan receive]:
testing.(*T).Run(0xc004142100, 0x59a989e, 0x1d, 0xc0035301b0, 0x9373be0)
	/usr/local/go/src/testing/testing.go:917 +0x693
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestRandomKeyAndTimestampExport(0xc004142100)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/export_test.go:429 +0x1041
testing.tRunner(0xc004142100, 0x5acf4b0)
	/usr/local/go/src/testing/testing.go:865 +0x164
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x65b

goroutine 2092 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
sync.runtime_notifyListWait(0xc000ac0130, 0x0)
	/usr/local/go/src/runtime/sema.go:510 +0xf9
sync.(*Cond).Wait(0xc000ac0120)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).syncLoop(0xc000ac0000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:693 +0x81
created by github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:680 +0x4e7

rax    0x0
rbx    0x7f5dcc3ef000
rcx    0x7f5dcb155428
rdx    0x6
rdi    0x6de0
rsi    0x6df8
rbp    0x695d378
rsp    0x7f5dc55fad98
r8     0xfffffffffffffff8
r9     0xfeff092d63646b68
r10    0x8
r11    0x202
r12    0x2d
r13    0x695d3f2
r14    0xc0036cc000
r15    0x2
rip    0x7f5dcb155428
rflags 0x202
cs     0x33
fs     0x0
gs     0x0



TestRandomKeyAndTimestampExport/ts_(0-∞],_all,_nontimebound
--- FAIL: testrace/TestRandomKeyAndTimestampExport/ts_(0-∞],_all,_nontimebound (0.000s)
Test ended in panic.




TestRandomKeyAndTimestampExport
--- FAIL: testrace/TestRandomKeyAndTimestampExport (0.000s)
Test ended in panic.

------- Stdout: -------
I190930 16:10:26.356912 2091 storage/engine/rocksdb.go:621  opening rocksdb instance at "/tmp/TestRandomKeyAndTimestampExport532539905"



TestRandomKeyAndTimestampExport/ts_(0-∞],_latest,_nontimebound
--- FAIL: testrace/TestRandomKeyAndTimestampExport/ts_(0-∞],_latest,_nontimebound (0.000s)
Test ended in panic.





```

Please assign, take a look and update the issue accordingly.

--- a/c-deps/libroach/incremental_iterator.cc
+++ b/c-deps/libroach/incremental_iterator.cc
@@ -42,7 +42,6 @@ DBIncrementalIterator::DBIncrementalIterator(DBEngine* engine, DBIterOptions opt
   // outside of the timestamp range **from iter's perspective**. This allows us
   // to simply ignore discrepancies that we notice in advance(). See #34819.
   if (!EmptyTimestamp(opts.min_timestamp_hint) || !EmptyTimestamp(opts.max_timestamp_hint)) {
-    assert(!EmptyTimestamp(opts.min_timestamp_hint));
     assert(!EmptyTimestamp(opts.max_timestamp_hint));
     DBIterOptions nontimebound_opts = DBIterOptions();
     nontimebound_opts.upper_bound = opts.upper_bound;

--- a/c-deps/libroach/iterator.cc
+++ b/c-deps/libroach/iterator.cc
@@ -26,7 +26,6 @@ DBIterator::DBIterator(std::atomic<int64_t>* iters, DBIterOptions iter_options) 
 
   if (!EmptyTimestamp(iter_options.min_timestamp_hint) ||
       !EmptyTimestamp(iter_options.max_timestamp_hint)) {
-    assert(!EmptyTimestamp(iter_options.min_timestamp_hint));
     assert(!EmptyTimestamp(iter_options.max_timestamp_hint));
     const std::string min = EncodeTimestamp(iter_options.min_timestamp_hint);
     const std::string max = EncodeTimestamp(iter_options.max_timestamp_hint);

